### ISSUE TITLE
Select only allows array of options

### ISSUE THREAD

**Issue by sp4cecat** [Shopify/polaris#174]
# Issue summary

There appears to be no way to use key/value pairs as <Select> options, only a list is allowed.

## Expected behavior

options should allow a list of objects, with each list item containing an 'id' and 'label' or perhaps a key/value relationship representing the options.

## Actual behavior

Throws an error if object is provided instead of array

## Steps to Reproduce the Problem

Insert a select with options as say:

[{ "QLD": "Queensland, "NSW": "New South Wales", .. }](url)

## Specifications

- Polaris version: 1.3.1
- React version: 15.6.1
- Browser:Any
- Device:Any
- Operating System:Any

**Comment by bakura10** [comment]
Hi,

You can do that by giving an array of hash this way:

```jsx
<Select
            label="Choose value"
            value={this.state.value}
            options={[
               {label: 'Foo', value: 'foo'},
               {label: 'Bar', value: 'bar'}
            ]}
          />
```

**Comment by lemonmade** [comment]
@sp4cecat @bakura10's way would be the recommended solution to this. Is there a reason you want to use an object instead of an array?

**Comment by sp4cecat** [comment]
It's not immediately obvious how you specify value:label pairs - the example at https://polaris.shopify.com/components/forms/select#navigation only shows a flat array for options.

**Comment by sp4cecat** [comment]
Ended up digging through the code to see the correct way; https://polaris.shopify.com/components is missing a bit of depth like that. The Select examples only show a flat array:
```
   options={[
    'two',
    'three',
  ]}
```
Is there another resource I can go to that's got a bit more detailed use info? To save me having to

**Comment by dfmcphee** [comment]
We are actively working on making improvements to our docs. I will add an issue to make sure we add some better and more complex examples for the `Select` component.

**Comment by dfmcphee** [comment]
Fixed in 1.6.0
